Code style improvements

awakeman opened the pull request:
<!-- PLEASE ENSURE YOU REVIEW THE CONTENT OF EACH PR CAREFULLY, INCLUDING SUBSEQUENT COMMENTS BY YOURSELF OR OTHERS. -->
<!-- IN PARTICULAR PLEASE ENSURE THAT SENSITIVE OR INAPPROPRIATE INFORMATION IS NOT UPLOADED -->
   
### Issues addressed by this PR
<!-- Add reference(s) to issue(s) solved by this PR. Please use keyword Fixes/Closes as per https://help.github.com/articles/closing-issues-using-keywords/ -->

Closes #175 
Closes #193 
<!-- Add short description of what has been fixed -->

- Moved all one-line if statement bodies to new line. (#175)
- Reordered Methods/Constructors/Properties/Fields into groups (#193)
- Added headers for groups (#193)
- Added method separators (#193)

### Test files
<!-- Link to test files to validate the proposed changes -->
N/A

### Changelog
<!-- Text to go into changelog if applicable -->
<!-- Please see https://github.com/BHoM/documentation/wiki/changelog for guidelines -->


### Additional comments
<!-- As required -->

adecler commented on Excel_UI/Addin/AddIn.cs:
```diff
@@ -356,70 +431,36 @@ private static bool IsNullMissingOrEmpty(object obj)
+        /******* Private fields                             **************/
+        /*****************************************************************/
+
+        private Dictionary<string, CallerFormula> m_formulea;
```
Incorrect variable naming. Please re-read details in issue #193

adecler replied:
Same for all other private variables

adecler commented on Excel_UI/UI/Callers/Explode.cs:
```diff
@@ -50,5 +46,15 @@ public ExplodeCaller() : base(typeof(Methods.Properties).GetMethod("Explode"))
         public override int GroupIndex => m_native.GroupIndex;
 
         private UI.Components.ExplodeCaller m_native = new UI.Components.ExplodeCaller();
+
```
Incorrect format for the properties

adecler commented on Excel_UI/UI/Components/Adapter/CreateAdapter.cs:
```diff
@@ -49,6 +49,8 @@ public class CreateAdapterFormula : CallerFormula
 
         public CreateAdapterFormula() : base() { }
+
+        /*******************************************/
```
- Incorrect format for first property

adecler replied:
Same in other files

awakeman commented:
Spoke with @al-fisher and we agreed it would be expedient to allow for someone other than @adecler to pick up this review in his absence as this is otherwise blocking other tasks on this toolkit. obviously reviewer should review against @adecler's comments to check they are appropriately addressed.

FraserGreenroyd commented on Excel_UI/UI/Templates/CallerFormula.cs:
```diff
@@ -35,25 +35,23 @@ namespace BH.UI.Excel.Templates
     public abstract class CallerFormula
     {
-        private IExcelSelectorMenu m_menu;
+        private IExcelSelectorMenu m_Menu;
```
Should this be at the bottom of the file?

awakeman commented:
> * In `AddIn.FlagUsed()`, `Wb` should be `wb`. Even better would be to use a more explicit name. Same in `Project.SaveData()`
Fixed
> * `Properties.Transpose()` has no separator and should in the private section.
Fixed
> * In `Properties.GetPropertyDictionaries()`, `else` should be on a new line. Same for `Project.GetId()` and `Project.Deserialise()`
Fixed
> * `Explode.m_Native` should be in the private field section
Fixed
> * `SingleOptionCallerFormula.MenuRoot` incorrect section and format.
Fixed
> * ArrayResizer`and`DebugConfig` not done
Fixed


> A few things on compliance as well. Although I am happy for those to be done on a separate PR as it was not part of this scope but thought I would mention it while I noticed them.
> 
> * Inconsistencies with file naming vs method naming (e.g. `FromExcel` vs `ToBHoM`, `Note` vs `SetNote`, `TextRef` vs `RefText`)
> * Incorrect naming of `Query` methods: should not start with `Get`
> * `Timer` should be in an `Objects` folder

This will be addressed in a compliance PR once this is merged
